### DISCUSSION TITLE
TCP JSON input Android

### DISCUSSION THREAD

**Discussion by Taxom** [jvde-github/AIS-catcher#167] [Ideas]
Can you add opportunity reception JSON messages on AIS Catcher Android from TCP AIS Catcher Debian?

**Comment by jvde-github** [comment]
Hi, there already is. You start your Debian instance with setting up a TCP listener/server like this:
```console
AIS-catcher -S 5012 JSON on
```
This will send NMEA packaged in JSON with meta data. You can receive how you normally listen to a NMEA TCP listener on Android (assuming your Debian sits at 192.168.1.113):

![image](https://github.com/jvde-github/AIS-catcher/assets/52420030/de75ec9f-0180-4f71-8f79-5c6f060a63ee)

**Reply by jvde-github** [reply]
Note that this JSON can only be read by AIS-catcher, so if you want to feed other applications with NMEA over TCP, you need to set up a second channel, e.g. -S 5011

**Comment by Taxom** [comment]
Yes, I'm sorry, this is my mistake. I changed the port but forgot to throw it through the NAT.  Everything works perfectly!

**Comment by Taxom** [comment]
The port forwarding to the NAT, without restart the Ais-Catcher revealed a strangeness. It stops sending data to marinetraffic and myshiptracking  (both TCP).  At the same time, there is nothing in the log of Ais-Catcher. Other UDP feeding aggregators is fine.

I think the router drops the connection during the conservation of the settings, but it is strange that the Ais-Catcher did not restore them.

![изображение](https://github.com/jvde-github/AIS-catcher/assets/20313019/c8dbe988-96ac-4ee8-87e9-754871e6f39d)
![изображение](https://github.com/jvde-github/AIS-catcher/assets/20313019/9cc9e7d1-edb2-4b02-85fa-831da1aa80d5)

**Reply by jvde-github** [reply]
Catcher will report errors and try to reconnect if there is an error. It reads like all existing TCP connections get confused because of this action without producing any errors in the sender and receiver. There is not much that you can about this, except for a periodic restart. The TCP client does not have a automatic restart option yet.
